dialog/file.go

```diff
@@ -208,13 +277,20 @@ func (f *fileDialog) refreshDir(dir string) {
 	f.fileScroll.Refresh()
 }
 
-func (f *fileDialog) setDirectory(dir string) {
+func (f *fileDialog) setDirectory(dir fyne.ListableURI) error {
+	if dir == nil {
+		return fmt.Errorf("failed to open nil directory")
+	}
+
 	f.setSelected(nil)
 	f.dir = dir
 
 	f.breadcrumb.Children = nil
-	buildDir := filepath.VolumeName(dir)
-	for i, d := range strings.Split(dir, string(filepath.Separator)) {
+
+	localdir := dir.String()[len(dir.Scheme())+3:]
+
+	buildDir := filepath.VolumeName(localdir)
+	for i, d := range strings.Split(localdir, "/") {
```

Comment:
We should consider a `.Components()` member for URI (or possibly just a `path` method which does this). We know that splitting on `/` is safe for URIs, but I worry that converting to a normal string and then doing the split later could end up causing confusion down the line. Not really applicable in this particular case, this is more of a general remark.

Reply:
This code will be stripped out. Generally it shouldn't be done like this on a URI and it is a hold-on from the old file code.
We have Child() and Parent() which supports traversing. Combined with the URIRootError we can figure out the whole path I am pretty sure. Addressing in follow-up PR